toolchain/check/eval.cpp

```diff
@@ -174,7 +177,8 @@ auto TryEvalInst(Context& context, SemIR::InstId inst_id, SemIR::Inst inst)
       value->value =
           (value->value == SemIR::BoolValue::False ? SemIR::BoolValue::True
                                                    : SemIR::BoolValue::False);
-      return context.AddConstantInst(*value);
+      return context.AddConstantInst(context.insts().GetParseNode(const_id),
```

Comment:
Note, this matches, but should it actually have been the inst_id's parse node?

Reply:
Yes, it probably should. However, I would like to remove the `NodeId` parameter from `AddConstantInst` entirely, as a follow-up after this PR lands. Because constant instructions get reused across different parse nodes (eg, currently, types get deduplicated, and eventually I'd like to deduplicate all constants), I think storing a parse node for them will invite bugs.